modules/imgproc/src/imgwarp.cpp

```diff
@@ -4288,10 +4290,32 @@ class RemapInvoker :
                             const float* sY = m2->ptr<float>(y+y1) + x;
                             x1 = 0;
 
-                        #if CV_SSE2
-                            if( useSIMD )
+                        #if CV_AVX2
+                            if (useAVX2)
                             {
-                                for( ; x1 <= bcols - 8; x1 += 8 )
+                                for (; x1 <= bcols - 16; x1 += 16)
+                                {
+                                    __m256 fx0 = _mm256_loadu_ps(sX + x1);
+                                    __m256 fx1 = _mm256_loadu_ps(sX + x1 + 8);
+                                    __m256 fy0 = _mm256_loadu_ps(sY + x1);
+                                    __m256 fy1 = _mm256_loadu_ps(sY + x1 + 8);
+                                    __m256i ix0 = _mm256_cvtps_epi32(fx0);
+                                    __m256i ix1 = _mm256_cvtps_epi32(fx1);
+                                    __m256i iy0 = _mm256_cvtps_epi32(fy0);
+                                    __m256i iy1 = _mm256_cvtps_epi32(fy1);
+                                    ix0 = _mm256_packs_epi32(ix0, ix1);
+                                    iy0 = _mm256_packs_epi32(iy0, iy1);
+                                    ix1 = _mm256_unpacklo_epi16(ix0, iy0);
+                                    iy1 = _mm256_unpackhi_epi16(ix0, iy0);
+                                    _mm256_storeu_si256((__m256i*)(XY + x1 * 2), ix1);
+                                    _mm256_storeu_si256((__m256i*)(XY + x1 * 2 + 16), iy1);
+                                }
+                                _mm256_zeroupper();
+                            }
+                        #elif CV_SSE2
```

Comment:
1. yes.
2. SSE & AVX are compatible, because they relate to x86/x64. But NEON is ARM specific and not compatible with SSE&AVX, so we use `elif` in this case.